R/mutate.R

```diff
@@ -0,0 +1,153 @@
+#' mutate for hyperSpec objects
+#'
+#' dplyr::mutate() adds mew variables and preserves all the existing variables.
+#' dplyr::transmute() adds and only keeps new variables.
+#' Special column `$spc` contains the spectra matrix.
+#' If `$spc` is not being transmuted, the result is a data.frame instead of a hyperSpec object.
+#'
+#' @inheritParams dplyr::mutate
+#' @inheritParams dplyr::transmute
+#' @seealso [dplyr::mutate()]
+#' @return hyperSpec object.
+#' @include unittest.R
+#' @include setLabels.R
+#' @importFrom hyperSpec chk.hy
+#' @importFrom hyperSpec labels labels<-
+#' @importFrom rlang quo_name
+#' @importFrom rlang quo_get_expr
+#' @importFrom dplyr mutate
+#' @importFrom dplyr transmute
+#'
+#' @examples
+
+#' # Mutate -----------------------------------------------
+#' chondro %>%
+#'   mutate (y, x, filename, spc) %>%
+#'   head # => results in a hyperSpec object
+#' chondro %>%
+#'   mutate (y, x)
+#'   head # => results in a hyperSpec object
+#' chondro %>%
+#'   mutate (y, x, filename, spc = spc*2)
+#'   head # => results in a hyperSpec object
+#' flu %>%
+#'   mutate (spc, filename, c)
+#'   head # => results in a hyperSpec object
+#' flu %>%
+#'   mutate (filename, c)
+#'   head # => results in a hyperSpec object
+#' flu %>%
+#'   mutate (c=c*2, c=c*0)
+#'   # => results in a hyperSpec object
+#'
+#' # Transmute -----------------------------------------------
+#' chondro %>%
+#'   transmute (y, x, filename, spc) %>%
+#'   head # => results in a hyperSpec object
+#' chondro %>%
+#'   transmute (y, x)
+#'   head # => results in a data frame
+#' chondro %>%
+#'   transmute (y, x, filename, spc = spc*2)
+#'   head # => results in a hyperSpec object
+#' flu %>%
+#'   transmute (spc, filename, c)
+#'   head # => results in a hyperSpec object
+#' flu %>%
+#'   transmute (filename, c)
+#'   head # => results in a data frame
+#' flu %>%
+#'   transmute (c=c*2, c=c*0)
+#'   # => results in a data frame
+#' @export
+mutate.hyperSpec <- function(.data, ...) {
+    # Check if user passed in a hyperSpec object
+    chk.hy(.data)
+    # Collect mutate arguments
+    ls_hy <- get_args(.data, ...)
```

Comment:
I find `ls_hy` a very unintuitive name for a variable that collects/contains arguments of a function call.

Reply:
I changed `ls_hy` to `mutate_args` or `transmute_args` depending on which is being called